File: testdata/bench/swift/BenchmarkTest00036.swift
import Vapor

func handler036(_ req: Request) throws -> Response {
    let input: String = try req.query.get(at: "input")
    let safe = input.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? ""
    let html = "<html><body><a href='/search?q=\(safe)'>Link</a></body></html>"
    return Response(status: .ok, body: .init(string: html))
}

File: testdata/bench/kotlin/BenchmarkTest00036.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import io.ktor.http.*

// Safe: escapeHTML utility function
fun escapeHtml(input: String?): String {
    return input?.replace("&", "&amp;")
        ?.replace("<", "&lt;")
        ?.replace(">", "&gt;")
        ?.replace("\"", "&quot;")
        ?.replace("'", "&#x27;") ?: ""
}

suspend fun handler00036(call: ApplicationCall) {
    val referer = call.request.headers["Referer"]
    call.respondText("<a href='${escapeHtml(referer)}'>Go back</a>", ContentType.Text.Html)
}
